examples/pod-security-groups/index.ts

```diff
@@ -0,0 +1,196 @@
+  cluster: cluster,
+  nodeRole: role,
+});
+
+const kube = cluster.provider;
+
+// Create a SecurityGroup for the nginx application
+const nginxSg = new aws.ec2.SecurityGroup("nginx", {
+  vpcId: eksVpc.vpcId,
+});
+// Allow all traffic between the cluster to the nginx SecurityGroup
+configureClusterAccess("nginx", cluster, nginxSg);
+// Create the nginx application (Deployment, Service, SecurityGroupPolicy)
+const nginx = app.createApplication("nginx", nginxSg.id, kube);
+
+/*
+ * Verify that the SecurityGroupPolicy is working as expected.
+ */
+
+// Create a SecurityGroup for the caller job
+const callerSg = new aws.ec2.SecurityGroup("caller", {
+  vpcId: eksVpc.vpcId,
+});
+// Allow all traffic between the cluster and the caller SecurityGroup
+configureClusterAccess("caller", cluster, callerSg);
+// Allow the caller job to access the nginx service
+new aws.vpc.SecurityGroupIngressRule("caller-to-nginx", {
+  securityGroupId: nginxSg.id,
+  ipProtocol: "tcp",
+  fromPort: 80,
+  toPort: 80,
+  referencedSecurityGroupId: callerSg.id,
+});
+
+// Assign the caller SecurityGroup to the caller job
+const callerSgp = new k8s.apiextensions.CustomResource("caller-sgp", {
+  apiVersion: "vpcresources.k8s.aws/v1beta1",
+  kind: "SecurityGroupPolicy",
+  metadata: {
+    name: "caller-sgp",
+  },
+  spec: {
+    podSelector: { matchLabels: { app: "caller" } },
+    securityGroups: {
+      groupIds: [callerSg.id],
+    }
+  },
+}, { provider: kube });
+
+// Create a job that is allowed to curl the nginx service. The job will fail if it can't reach the service.
+new k8s.batch.v1.Job("caller", {
+  spec: {
+    template: {
+      metadata: {
+        name: "caller",
+        labels: {
+          app: "caller",
+        },
+      },
+      spec: {
+        containers: [{
+          name: "caller",
+          image: "curlimages/curl",
+          command: ["curl", "--silent", "--show-error", "--fail", pulumi.interpolate`${nginx.metadata.name}.${nginx.metadata.namespace}:80`],
+        }],
+        restartPolicy: "Never",
+      },
+    },
+    backoffLimit: 3,
+  },
+}, { provider: kube, dependsOn: [nginx, callerSgp] });
+
+// Create a SecurityGroup for the wrongCaller job
+const wrongCallerSg = new aws.ec2.SecurityGroup("wrong-caller", {
+  vpcId: eksVpc.vpcId,
+});
+// Allow all traffic between the cluster and the caller SecurityGroup
+configureClusterAccess("wrong-caller", cluster, wrongCallerSg);
+
+// Assign the caller SecurityGroup to the wrongCaller job
+const wrongCallerSgp = new k8s.apiextensions.CustomResource("wrong-caller-sgp", {
+  apiVersion: "vpcresources.k8s.aws/v1beta1",
+  kind: "SecurityGroupPolicy",
+  metadata: {
+    name: "wrong-caller-sgp",
+  },
+  spec: {
+    podSelector: { matchLabels: { app: "wrong-caller" } },
+    securityGroups: {
+      groupIds: [wrongCallerSg.id],
+    }
+  },
+}, { provider: kube });
+
+// Create a job that is not allowed to curl the nginx service. The job will fail if it can reach the service.
```

Comment:
This feels like a counter-example to test something instead of a best-practice example of how to GTD, or am I misunderstanding? Does it belong under ./examples?

Reply:
I'd say showing that traffic can be restricted is an important part of a firewall example.

Reply:
Hmm, yeah I concede the point. A few comments to that end then perhaps? Maybe in the readme? I still get confused by mixing tests and examples in one.